git-repository/src/env.rs

```diff
@@ -34,3 +34,86 @@ pub fn args_os() -> impl Iterator<Item = OsString> {
 pub fn os_str_to_bstring(input: &OsStr) -> Option<BString> {
     Vec::from_os_string(input.into()).map(Into::into).ok()
 }
+
+/// Utilities to collate errors of common operations into one error type.
+///
+/// This is useful as this type can present an API to answer common questions, like whether a network request seems to have failed
+/// spuriously or if the underlying repository seems to be corrupted.
+/// Error collation supports all operations, including opening the repository.
+///
+/// ### Usage
+///
+/// The caller may define a function that specifies the result type as `Result<T, git::env::collate::<operation>::Error>` to collect
```

Comment:
```suggestion
/// The caller may define a function that specifies the result type as `Result<T, git::env::collate::{operation}::Error>` to collect
```
maybe? since `collate::<operatoion>` could be a turbofish?

Reply:
That's a great point! It's changed (next time I push).